src/main/java/org/testeditor/web/xtext/index/FileBasedXtextIndexFiller.xtend

```diff
@@ -0,0 +1,24 @@
+package org.testeditor.web.xtext.index
+
+import java.io.File
+import org.eclipse.emf.common.util.URI
+
+class FileBasedXtextIndexFiller {
+	
+	/**
+	 * redefine to match only files of your languages to be added to the index
+	 */
+	def boolean isIndexRelevant(File file) {
+		return true
+	}
+	
+	def void fillWithFileRecursively(XtextIndex index, File file) {
+		file.listFiles?.forEach[
+			fillWithFileRecursively(index, it)
```

Comment:
If you swap the arguments of the method, you should be able to use it extension-method-style and avoid the `it`: `file.fillWithFileRecursively(index)`.
(Not sure if that would be an improvement though...)

Reply:
yeah, but index is the object that is actually worked on, so I wanted to have it in first position.